.github/workflows/downstream.yml

```
@@ -0,0 +1,46 @@
+name: downstream
+on:
+  push:
+    branches:
+      - main
+    tags: '*'
+  pull_request:
+
+# Needed to allow julia-actions/cache to delete old caches that it has created
+permissions:
+  actions: write
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  test:
+    name: downstream ${{ matrix.package }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        package:
+          - 'ClimaCore.jl'
+        version:
+          - '1.10'
+          - '1.11'
+    steps:
+      - uses: actions/checkout@v4
+      - uses: julia-actions/setup-julia@v2
+        with:
+          version: ${{ matrix.version }}
+      - uses: julia-actions/cache@v2
+      - uses: julia-actions/julia-buildpkg@v1
+      - uses: actions/checkout@v4
+        with:
+          repository: 'CliMA/${{ matrix.package }}'
+          path: ${{ matrix.package }}
+
+      - run: |
+          julia --color=yes -e 'using Pkg; Pkg.add("TestEnv")'
+          julia --color=yes --project=${{ matrix.package }} -e 'using Pkg; Pkg.instantiate()'
+          julia --color=yes --check-bounds=yes --project=${{ matrix.package }} -e 'using TestEnv; TestEnv.activate();\
+            using Pkg; Pkg.develop(; path = "."); include("${{ matrix.package }}/test/runtests.jl")'
```

```suggestion
          julia --color=yes --project=${{ matrix.package }} -e 'using Pkg; Pkg.instantiate(); Pkg.develop(; path = "."); Pkg.test()'
```
This can be simplified a bit.

---

Fine by me.